app/src/main/java/org/mozilla/fenix/library/history/HistoryAdapter.kt

```diff
@@ -0,0 +1,221 @@
+                    actionEmitter.onNext(HistoryAction.Delete.Some(mode.selectedItems))
+                } else {
+                    actionEmitter.onNext(HistoryAction.Delete.All)
+                }
+            }
+        }
+
+        private val text = view.findViewById<TextView>(R.id.delete_history_button_text).apply {
+            val color = ContextCompat.getColor(context, R.color.photonRed60)
+            val drawable = ContextCompat.getDrawable(context, R.drawable.ic_delete)
+            drawable?.setTint(color)
+            this.setCompoundDrawablesWithIntrinsicBounds(drawable, null, null, null)
+        }
+
+        fun bind(mode: HistoryState.Mode) {
+            this.mode = mode
+
+            val text = if (mode is HistoryState.Mode.Editing && mode.selectedItems.isNotEmpty()) {
+                text.context.resources.getString(R.string.history_delete_some, mode.selectedItems.size)
+            } else {
+                text.context.resources.getString(R.string.history_delete_all)
+            }
+
+            button.contentDescription = text
+            this.text.text = text
+        }
+
+        companion object {
+            const val LAYOUT_ID = R.layout.history_delete
+        }
+    }
+
+    private var items: List<HistoryItem> = emptyList()
+    private var mode: HistoryState.Mode = HistoryState.Mode.Normal
+
+    fun updateData(items: List<HistoryItem>, mode: HistoryState.Mode) {
+        this.items = items
+        this.mode = mode
+        notifyDataSetChanged()
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
+        val view = LayoutInflater.from(parent.context).inflate(viewType, parent, false)
+
+        return when (viewType) {
+            HistoryDeleteViewHolder.LAYOUT_ID -> HistoryDeleteViewHolder(view, actionEmitter)
+            HistoryHeaderViewHolder.LAYOUT_ID -> HistoryHeaderViewHolder(view)
+            HistoryListItemViewHolder.LAYOUT_ID -> HistoryListItemViewHolder(view, actionEmitter)
+            else -> throw IllegalStateException("viewType $viewType does not match to a ViewHolder")
+        }
+    }
+
+    override fun getItemViewType(position: Int): Int {
+        return when (position) {
+            0 -> HistoryDeleteViewHolder.LAYOUT_ID
+            1 -> HistoryHeaderViewHolder.LAYOUT_ID
+            else -> HistoryListItemViewHolder.LAYOUT_ID
+        }
+    }
+
+    override fun getItemCount(): Int = items.count() + NUMBER_OF_SECTIONS
+
+    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
+        when (holder) {
+            is HistoryDeleteViewHolder -> holder.bind(mode)
+            is HistoryHeaderViewHolder -> holder.bind("Today")
+            is HistoryListItemViewHolder -> holder.bind(items[position - NUMBER_OF_SECTIONS], mode)
+        }
+    }
+
+    companion object {
+        private const val NUMBER_OF_SECTIONS = 2
```

Comment:
This is temporary until we extract time based sections into a reusable component